### ISSUE TITLE
Allow automatic install/setup of a required java version

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#938]
Currently if I import a project with a JVM that is not installed (lets say I have only installed java 11 but project has java = 17) JDT simply uses the older one (what of course will not work).

What I need to do is

1. Get the JVM from somwhere
2. Goto Java Settings and add it to the list of installed JDKs

It would be great if m2e could detect this issue and then ask if it should download and configure suitable JVM. We could use JustJ for that the following way:

1. Query the JustJ site for a suitable java version
2. Download it to the local maven repository (@merks is JustJ also deployed to maven central?), or use a previously downloaded one
3. extract it and use that as a location to configure JDT with an additional JVM

This is a bit a counter part of the Tycho issue:
- https://github.com/eclipse-tycho/tycho/issues/1163

**Comment by HannesWell** [comment]
That's a good suggestion, but isn't that interesting for Plugin Projects and even plain Java projects as well?
So maybe it would be better to place that in JDT?

**Comment by laeubi** [comment]
> That's a good suggestion, but isn't that interesting for Plugin Projects and even plain Java projects as well?
> So maybe it would be better to place that in JDT?

Maybe yes, I just noticed that in the context of m2e but for sure it could be solved more gneral in JDT (if there is any one confident to do it).

**Comment by mickaelistria** [comment]
I agree it's a more general problem and that JDT is a better target.
